packages/amplify_core/lib/src/types/query/query_field_operators.dart

```diff
@@ -79,6 +79,22 @@ abstract class QueryFieldOperator<T> {
     // TODO sanitize other types appropriately
     return value;
   }
+
+  /// Gets the correct field to use as a comparison for nested models.
+  ///
+  /// TODO: Remove when codegen supports a `toMap()` method.
+  Object? _getOtherField(Object? other, Object? value) {
+    // if `other` is type Map and `value` is type String, the
+    // only valid query predicate is for a nested model by ID.
+    //
+    // For example: `Comment.POST.eq('post-id-123')`
+    //
+    // Return the ID to use as the comparison field.
+    if (other is Map && value is String) {
+      return other['id'];
```

Comment:
Does this work with CPK?

Reply:
No. It already does not work with CPK though. We need code changes to support this with CPK. We have no way to get the `ModelIdentifier` of the nested model without reflection in Dart, or a codegen change to fill the gap (i.e. a `toMap()` method). This is because the query predicate just has the string field name (`"blog"` for a predicate of `Post.BLOG.eq("blog-id")`).

Those codegen changes will be non-breaking, but consuming them will be breaking (customers will need to regenerate models) unfortunately. 

I think it is debatable if it is worth moving forward with this. The benefit would be to get this fixed for those not using CPK before we 1) get the codgen change in, and 2) do out next breaking change.

If we can do 1 & 2 somewhat quickly, we could just opt to fix this the right way from the start.

Reply:
@HuiSF - Curious to get your thoughts on this. I don't think the codegen changes will be too complicated, but I haven't had a chance to look at them yet.

Reply:
Support of creating predicate on nested model is a bit wonky :/ The current implementation is a bit hacky already to adapt to different patterns between amplify-ios and amplify-android. I think it makes sense to make codegen necessary codegen change to make this working.

Re: the breaking change. We've added a deprecation warning on `Model.getId()` method, that after upgrading amplify-flutter library 0.6.6, this warning gets shown when build App. Hopefully this helps making the migration smoother.

Reply:
Okay. I think that is fair. I can look into the codegen changes. Will mark this PR as draft until then.